File: Wolf Beacon/Wolf Beacon/Views/Cells/BaseTableViewCell.m
//
//  BaseTableViewCell.m
//  Wolf Beacon
//
//  Created by Avikant Saini on 10/26/16.
//  Copyright © 2016 Wolf Beacon. All rights reserved.
//

#import "BaseTableViewCell.h"

@implementation BaseTableViewCell

- (void)awakeFromNib {
    [super awakeFromNib];
    // Initialization code
	
	self.topSeparator = NO;
	self.topOffset = 20.f;
	
	self.bottomSeparator = NO;
	self.bottomOffset = 20.f;
	
}

- (void)drawRect:(CGRect)rect {
	[super drawRect:rect];
	
	CGRect bounds = self.bounds;
	CGSize size = bounds.size;
	
	if (self.topSeparator) {
		UIBezierPath *beizerPath = [UIBezierPath new];
		[beizerPath moveToPoint:(CGPointMake(self.topOffset, 0))];
		[beizerPath addLineToPoint:(CGPointMake(size.width, 0))];
		beizerPath.lineWidth = 0.5;
		[UIColor.lightGrayColor setStroke];
		[beizerPath stroke];
	}
	
	if (self.bottomOffset) {
		UIBezierPath *beizerPath = [UIBezierPath new];
		[beizerPath moveToPoint:(CGPointMake(self.topOffset, size.height - 0.5))];
		[beizerPath addLineToPoint:(CGPointMake(size.width, size.height - 0.5))];
		beizerPath.lineWidth = 0.5;
		[UIColor.lightGrayColor setStroke];
		[beizerPath stroke];
	}
	
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];

    // Configure the view for the selected state
}

@end

File: Wolf Beacon/Wolf Beacon/Views/Cells/BaseTableViewCell.swift
//
//  BaseTableViewCell.swift
//  Wolf Beacon
//
//  Created by Avikant Saini on 5/21/16.
//  Copyright © 2016 Open Source Hacking. All rights reserved.
//

import UIKit

class BaseTableViewCell: UITableViewCell {
	
	var topSeparator: Bool = false {
		didSet {
			self.setNeedsDisplay()
		}
	}
	
	var topOffset: CGFloat = 0.0 {
		didSet {
			self.setNeedsDisplay()
		}
	}
	
	var bottomSeparator: Bool = false {
		didSet {
			self.setNeedsDisplay()
		}
	}
	
	var bottomOffset: CGFloat = 0.0 {
		didSet {
			self.setNeedsDisplay()
		}
	}
	
	override func drawRect(rect: CGRect) {
		super.drawRect(rect)
		let bounds = self.bounds
		let size = bounds.size
		if (topSeparator) {
			let beizerPath = UIBezierPath()
			beizerPath.moveToPoint(CGPointMake(topOffset, 0))
			beizerPath.addLineToPoint(CGPointMake(size.width - topOffset, 0))
			beizerPath.lineWidth = 0.5
			UIColor.lightGrayColor().setStroke()
			beizerPath.stroke()
		}
		if (bottomSeparator) {
			let beizerPath = UIBezierPath()
			beizerPath.moveToPoint(CGPointMake(topOffset, size.height - 0.5))
			beizerPath.addLineToPoint(CGPointMake(size.width - topOffset, size.height - 0.5))
			beizerPath.lineWidth = 0.5
			UIColor.lightGrayColor().setStroke()
			beizerPath.stroke()
		}
	}

    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
    }

    override func setSelected(selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)

        // Configure the view for the selected state
    }

}
